document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -3362,6 +3364,37 @@ public void testMongoFunctionExpressionGroupBy(String dataStoreName) throws Exce
     testCountApi(dataStoreName, query, "query/function_expression_group_by_response.json");
   }
 
+  @ParameterizedTest
+  @ArgumentsSource(MongoProvider.class)
+  public void testToLowerCaseMongoFunctionOperator(String dataStoreName) throws Exception {
+    Collection collection = getCollection(dataStoreName);
+
+    List<SelectionSpec> selectionSpecs =
+        List.of(
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("price")),
+            SelectionSpec.of(IdentifierExpression.of("quantity")),
+            SelectionSpec.of(IdentifierExpression.of("date")));
+    Selection selection = Selection.builder().selectionSpecs(selectionSpecs).build();
+    Filter filter =
+        Filter.builder()
+            .expression(
+                RelationalExpression.of(
+                    FunctionExpression.builder()
+                        .operator(FunctionOperator.TO_LOWER_CASE)
+                        .operand(IdentifierExpression.of("item"))
+                        .build(),
+                    RelationalOperator.EQ,
+                    ConstantExpression.of("shampoo")))
+            .build();
+
+    Query query = Query.builder().setSelection(selection).setFilter(filter).build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsAndSizeEqualWithoutOrder(
+        dataStoreName, resultDocs, "query/simple_filter_response.json", 2);
```

Comment:
`simple_filter_response.json` seem to contain 5 items and seem to contain items other than "shampoo" as well.